### PR TITLE
Tweaks and improvements to file handling

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary
* Handles corrupt files better during download by raising a handled exception rather than assertion error
* Handles URLs with query and hash parameters when extracting extensions

## References
Issues discovered during https://github.com/learningequality/kolibri-library/issues/19

## Reviewer guidance
Read tests carefully and ensure they are doing what they assert - they were generated by AI, and while I have read through them to ensure the same, a second look is helpful.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved error handling for empty temporary files by raising a specific exception instead of a generic assertion failure.
  - Enhanced file extension extraction to better handle URLs, query strings, and edge cases, ensuring consistent and accurate results.

- **Tests**
  - Added comprehensive unit tests for file extension extraction, covering various file paths, URLs, and edge cases.

### CODE REVIEW

**Review by marcellamaki — COMMENTED** [review]
The tests all seem to align with my understanding of them. Added just a couple of comments just for clarity

File: tests/utils/test_extensions.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import pytest
+
+from ricecooker.utils.utils import extract_path_ext
+
+# Tests generated by Claude Sonnet 3.7
```
[the remark this answers is not in the document]
Comment (rtibbles) [reply tests/utils/test_extensions.py:5]: This is the first time I've explicitly added it, at least partly because it's the first time I have used tests verbatim from LLM output, because they were good enough.

File: tests/utils/test_extensions.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    assert extract_path_ext("ftp://files.net/downloads/file.zip") == "zip"
+
+
+def test_extract_path_ext_urls_with_query_params():
+    """Test URLs with query parameters."""
+    assert extract_path_ext("http://example.com/file.html?param=value") == "html"
+    assert (
+        extract_path_ext("https://api.domain.org/data.json?id=123&token=abc") == "json"
+    )
+    assert extract_path_ext("http://site.com/download.tar.gz?download=true") == "gz"
```
[the remark this answers is not in the document]
Comment (rtibbles) [reply tests/utils/test_extensions.py:36]: I think generally the convention is that a TAR file by itself is not compressed, it is just an archive of files, so that would just be `.tar` if it is then also compressed you add the extension `.gz`. So if we were dealing with these types of files in ricecooker, we would use the new file pipeline to handle the `.gz` file extension, then spit out a `.tar` file that would be handled by another handler.

Until we do need to handle them though, I think this is mostly academic.

**Review by marcellamaki — APPROVED** [review]
